ledger/participant-state/kvutils/tools/src/test/scala/ledger/participant/state/kvutils/tools/integritycheck/IntegrityCheckerSpec.scala

```diff
@@ -163,5 +164,5 @@ final class IntegrityCheckerSpec
   }
 
   private def countOccurrences(input: String, pattern: String): Int =
-    input.sliding(pattern.length).count(_ == pattern)
+    new WrappedString(input).sliding(pattern.length).map(_.toString).count(_ == pattern)
```

Comment:
this is a bit of a mess. On 2.13, `sliding` on string is deprecated. `.toSeq.sliding` works but doesn’t work in 2.12. This convoluted mess works in both.

Reply:
How about `Pattern.quote(pattern).split(input).length - 1`?

Probably marginally less efficient, but this is a test, so :shrug: .

Reply:
That’s not equivalent for overlapping patterns but I guess we don’t care about that, fixed in 2a3c37c59920cf8a2ee7b6dbf4b10093183f91e8

Reply:
actually this is a mess. we rely on matches at the end afaict because we match on line endings. While we probably can fix this to make it work with a regex match, at that point it seems worse than this.

Reply:
Well, splitting should handle that (you'll get an empty string at the end of the array), but I'm happy with this solution too if you like.